lib/fileHandler/index.js

```diff
@@ -104,9 +117,142 @@ module.exports = function (options) {
 
 
 	/*
-  * Download a file from S3. The object passed here should have been one that
-  * was returned via the `upload` function above in a previous workflow step.
-  */
+		A stream variant of upload, requiring a `readStream` property
+		instead of a `file` property
+	*/
+	var streamUpload = function (params) {
+		return when.promise(function (resolve, reject) {
+
+			if (!_.isObject(options.aws)) {
+				return reject(new Error('AWS config not specified. Add a aws.json file in the connector folder, and pass it to falafel.'));
+			}
+
+			if (_.isUndefined(params.readStream) || _.isUndefined(params.readStream.pipe)) {
+				return reject(new Error('The object passed in must contain the property \'readStream\', referecing a read stream.'));
+			}
+
+			var client = knox.createClient({
+				key: options.aws.key,
+				secret: options.aws.secret,
+				bucket: 'workflow-file-uploads',
+				region: 'us-west-2'
+			});
+
+
+			// Generate a random file name
+			var fileName = guid();
+
+			var mimeType = params.contentType || mime.lookup(params.name);
+
+			var req = client.put(fileName, {
+				'Content-Length': params.length,
+				'Content-Type': mimeType,
+			});
+
+			params.readStream.pipe(req);
+
+			req.on('response', function(res){
+				if (res.statusCode !== 200) {
+					reject(new Error(res.statusCode));
+				} else {
+					var twentyFourHours = new Date(Date.now() + 24 * 60 * 60000);
+					var signedUrl = client.signedUrl(fileName, twentyFourHours);
+
+					// Resolve with an object following the "file" schema
+					resolve({
+						name: params.name,
+						url: signedUrl,
+						mime_type: mimeType,
+						expires: Math.round(twentyFourHours.getTime() / 1000)
+					});
+				}
+			});
+
+			req.on('error', reject);
```

Comment:
are you sure the error is an Error? i've seen in some other places dealing with files that it is a string, so then the backend receives a wrong format. Would be best to actually handle it globally in falafel for all connectors - if rejected thing is a string, wrap it in an Error

Reply:
Ah, keep in mind that these file helpers are not connector operations themselves, but used with operations. So the operation itself will decide how to handle the error, whatever the error is.